flake.nix

```
@@ -1,23 +1,70 @@
 {
-  description = "nickel-kubernetes development environment";
+  description = "nickel-kubernetes: Nickel libraries and tools to write Kubernetes resources natively in Nickel";
 
   inputs = {
     flake-utils.url = "github:numtide/flake-utils";
-    pkgs.url = "github:NixOS/nixpkgs/nixos-21.05";
-    nickel.url = "github:tweag/nickel";
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-23.11";
+    json-schema-to-nickel.url = "github:nickel-lang/json-schema-to-nickel";
   };
 
-  outputs = { self, nickel, pkgs, flake-utils }:
-    flake-utils.lib.eachDefaultSystem (system: {
-      defaultPackage = (import ./default.nix {
-        system = system;
-        pkgs = pkgs.legacyPackages.${system};
-        nickel = nickel.defaultPackage.${system} or null;
-      }).nickel_defs;
-      devShell = (import ./default.nix {
-        system = system;
-        pkgs = pkgs.legacyPackages.${system};
-        nickel = nickel.defaultPackage.${system} or null;
-      }).shell;
-    });
+  outputs = { self, nixpkgs, flake-utils, ... }@inputs:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        json-schema-to-nickel =
+          inputs.json-schema-to-nickel.packages.${system}.default;
+
+        pkgs = import nixpkgs { inherit system; };
+
+        # K8s JSON schemas of a specific K8s version taken from
+        # `kubernetes-json-schema` (as a path)
+        k8sSchemasRepo = pkgs.fetchFromGitHub {
+          owner = "yannh";
+          repo = "kubernetes-json-schema";
+          rev = "f8b907d7210830dfecbf976c2ba3c849bcf0cace";
+          hash = "sha256-Vd3iHeB7LvFDNu+m/RE2P6zgRhNJemiE+j1AAuc6swo=";
+        };
+
+        k8sSchemas = { version ? latestK8sVersion }: k8sSchemasRepo + "/${version}";
```

```suggestion
        k8sSchemas = { version ? latestK8sVersion }: "${k8sSchemasRepo}/${version}";
```
Caution: this might yield a string instead of a path, I'm not quite sure. But I hate `+` for string concatenation on commutativity grounds :sweat_smile:

---

I believe `path + string` is the idiomatic way in Nix to ensure that the end result is a path indeed. I've been bitten before by having a string instead of a path. `builtins.toPath` used to exist, but it's been replaced by just using `path + string` as well (see e.g. https://github.com/NixOS/nixpkgs/pull/40937). Unless there's a solid way to ensure the result is a path, I would prefer to stick to the current version